### ISSUE TITLE
Check if illegal options are specified for the different commands

### ISSUE THREAD

**Issue by torognes** [torognes/vsearch#37]

**Comment by frederic-mahe** [comment]
Torbjørn is going to create lists of legal options for each command and add tests to detect the usage of illegal options. vsearch will print an error message with the illegal option,  the list of legal options for that particular command, and the short general help listing all the commands.

**Comment by colinbrislawn** [comment]
This is helpful, especially when users pass flags not yet implemented in vsearch.

For example, if I ran
`vsearch --cluster_fast seqs.derep.fna --abskew 10 ...`
I could reasonably expect vsearch was using the abundance skew threshold for clustering, when vsearch is actually ignoring the unrecognized flag.

**Comment by torognes** [comment]
Commit bb3c2d435f0fff26b874771e42d46ce8d74be99a : valid options are now checked.

There might be some doubt about which options are allowed for each command. Some might be allowed even if they have no effect or even if they are not documented.

**Comment by torognes** [comment]
VSEARCH will now fail if an unsupported option is specified for the command given. It will also list the available options for that command. Example:

```
$ vsearch --shuffle input.fasta --self --threads 8
Fatal error: Invalid options to command shuffle
Invalid option(s): --self --threads
The valid options for the shuffle command are: --bzip2_decompress --fasta_width --fastq_ascii --fastq_qmax --fastq_qmin --gzip_decompress --log --maxseqlength --minseqlength --no_progress --notrunclabels --output --quiet --randseed --relabel --relabel_keep --relabel_md5 --relabel_sha1 --sizeout --topn --xee --xsize
```

**Comment by torognes** [comment]
Fixed in vsearch version 2.13.0 just released.

**Comment by torognes** [comment]
A user complained that it was no long possible to use the `--threads` option with the `fastq_eestats2` command, even with `1` as the argument. Although many commands do not support multithreading, we should perhaps allow this option anyway with all commands? Some other options (e.g. `--log`, `--quiet`, `--no_progress`) should perhaps also be allowed with all commands even if they do not have a function.

**Comment by frederic-mahe** [comment]
I see. A more stringent option check was very likely to break some pipelines. The problem of silently allowing options without any effect is that it may create false beliefs: for example, command `fastq_eestats2` is multithreaded (it is not).

Maybe we could meet users halfway and allow the general options you've listed, but emit a warning if it has no effect?

For example:
```
Warning: option --threads has no effect on command --fastq_eestats2
```

The options associated to a given command would then be classified as: 
- applicable to the command (white list),
- not applicable but tolerated with a warning (gray list),
- not applicable (all the options not in the white or gray lists)

**Comment by torognes** [comment]
I agree. A warning would be appropriate in this case. I am afraid many pipelines would break if we're too strict.

**Comment by colinbrislawn** [comment]
I like the idea of showing a warning, and I even support throwing an error with non-functional flags like `--threads` and `--log`. I worry that folks will use non-flags and then wonder "where's my log file??"

While some pipelines make break, I think that 1) removing a flag is an easy fix and 2) this could be a good wake-up call to developers who thought a command had threads when it does not.

**Comment by torognes** [comment]
Regarding the `--threads` option: Since it is technically correct to use `--threads 1` for single-threaded commands, perhaps we could allow the option for all commands, but give a warning or an error if it is used with an argument larger than 1 for those commands that do not support multithreading?

**Comment by frederic-mahe** [comment]
@torognes I agree, but that's extra-conditional coding. Up to you since you are the one implementing all this :-)

To summarize, vsearch sends a warning when general options (`--log`, `--quiet`, `--no_progress`) or `--threads n` (n > 1) are used with a command that is not affected by them.

**Comment by torognes** [comment]
I've chosen to allow `--quiet`, `--log` and `--threads` for all commands. The first two always have an effect. The latter will generate a warning for the value is higher that 1 for commands that are not multi-threaded. Implemented in commit 37fa9037280b1714cfb0a4250da9803a7e7a548c.

**Comment by torognes** [comment]
Fixed in VSEARCH 2.13.1, just released.
